sub-packages/bionemo-noodles/src/bionemo/noodles/nvfaidx.py

```diff
@@ -0,0 +1,112 @@
+from noodles_fasta_wrapper import IndexedFastaReader
+
+class SequenceAccessor:
+    # NOTE: we could totally handle this stuff in Rust if we want.
+    def __init__(self, reader, seqid, length):
+        self.reader = reader
+        self.seqid = seqid
+        self.length = length
+
+    def __getitem__(self, key):
```

Comment:
We should be clear about how we're 0 based in docs. Maybe make a docstring for the class and spell that out.